jina/peapods/__init__.py

```diff
@@ -76,24 +72,51 @@ def Pod(args: Union['argparse.Namespace', Dict] = None, allow_remote: bool = Tru
                 from .pods.mutable import MutablePod
                 return MutablePod(args)
             else:
-                # TODO: this part needs to be refactored
-                from .pods.remote import RemoteMutablePod
-                return RemoteMutablePod(args)
+                from .runtimes.remote.jinad import RemoteJinaDRunTime
+                return RemoteJinaDRunTime(args, kind='pod')
 
     if not allow_remote and args.host != __default_host__:
         args.host = __default_host__
         default_logger.warning(f'host is reset to {__default_host__} as allow_remote=False')
 
     if args.host != __default_host__:
         if args.remote_access == RemoteAccessType.JINAD:
-            from .pods.remote import RemotePod
-            return RemotePod(args)
+            from .runtimes.remote.jinad import RemoteJinaDRunTime
+            return RemoteJinaDRunTime(args, kind='pod')
         elif args.remote_access == RemoteAccessType.SSH:
-            from .peas.remote.ssh import RemoteSSHPod
-            return RemoteSSHPod(args)
+            from .runtimes.remote.ssh import RemoteSSHRunTime
+            return RemoteSSHRunTime(args, kind='pod')
         else:
             raise ValueError(f'{args.remote_access} is not supported')
-
     else:
         from .pods import BasePod
         return BasePod(args)
+
+
+def Pea(args: Optional['argparse.Namespace'] = None,
+        ctrl_addr: Optional[str] = None,
+        ctrl_with_ipc: Optional[bool] = None,
+        gateway: bool = False,
+        rest_api: bool = False):
```

Comment:
All other arguments except `args` are gateway specific. Following jina's pattern, let's pass them using `**kwargs`

Reply:
okk

Reply:
well `gateway` and `rest_api` are needed in this specific context. ctrl_addr and ctrl_with_ipc sure I will change them